packages/react-server/src/ReactFlightServer.js

```diff
@@ -3892,9 +3926,18 @@ function emitChunk(
     return;
   }
   // For anything else we need to try to serialize it using JSON.
-  // $FlowFixMe[incompatible-type] stringify can return null for undefined but we never do
-  const json: string = stringify(value, task.toJSON);
-  emitModelChunk(request, task.id, json);
+  // We stash the outer parent size so we can restore it when we exit.
+  // We don't reset the serialized size counter from reentry because that indicates that we
+  // are outlining a model and we actually want to include that size into the parent since
+  // it will still block the parent row. It only restores to zero at the top of the stack.
+  const parentSerializedSize = 0;
```

Comment:
I'm probably misinterpreting this comment, but to me "We stash the outer parent size so we can restore it when we exit" sounds like `const parentSerializedSize = serializedSize;` here, along with `serializedSize = parentSerializedSize` in `finally`. But then there would need to be `serializedSize = 0` somewhere else to match "It only restores to zero at the top of the stack". Which is wrong, a) the comment, b) the code, or c) my interpretation?

Reply:
The code is wrong. Not sure how that happened. Fixed.

https://github.com/facebook/react/pull/33030/commits/54104be413157fb8355c49e28153d0a73b4b32cf

It shouldn't set `serializedSize = 0`, as the comment implies, because we assume that synchronously outlined models are actually conceptually part of the current serialization since they'll block the current row too.